### ISSUE TITLE
[ObjC] Generated code with polymorphism duplicates properties in subclass 

### ISSUE THREAD

**Issue by Dreddik** [swagger-api/swagger-codegen#4128]
##### Description

Generated objective-c code for polymorph classes duplicates obj-c properties in base class and in subclass. This cause a ton of warnings:
```
Auto property synthesis will not synthesize property '_id'; it will be implemented by its superclass, use @dynamic to acknowledge intention
```

##### Swagger-codegen version

2.2.1

##### Swagger declaration file content or url
```yaml
swagger: '2.0'

info:
  version: "1"
  title: "Hello"

basePath : "/api/v1"
produces:
  - application/json
schemes:
  - https
        
paths:
  /users:
    get:
      description: get ssers
      responses:
        200:
          description: Successful response
          schema:
            type: array
            items:
              $ref: '#/definitions/UserBasic'
  /advisors:
    get:
      description: get advisors
      responses:
        200:
          description: Successful response
          schema:
            type: array
            items:
              $ref: '#/definitions/UserAdvisor'
            
        
definitions:
  UserBasic:
    description: Basic information about user in API
    type: object
    discriminator: id
    required:
      - id
      - fullname
      - type
    properties:
      id:
        type: integer
        format: int64
      type:
        type: string
        default: common
        enum:
          - common
          - advisor
      fullname:
        description: User must specify his full name when sign up
        type: string
      online:
        description: Flag marks is user online or not
        type: boolean
      languages:
        description: Languages user speak (localized value)
        type: array
        items:
          type: string
  
  UserAdvisor:
    description: A user who can provide consultation service
    allOf:
      - $ref: '#/definitions/UserBasic'
      - type: object
        properties: 
          available_now:
            type: boolean
  
```

##### Command line used for generation

swagger-codegen generate -i ./swagger.yaml -l objc -o ./SwaggerClient

##### Steps to reproduce

1) Generate code
2) Create a test project, create Podfile with contents:
```
use_frameworks!
platform :ios, '9.0'

target 'TestSwagger' do
    pod 'SwaggerClient', :path => './SwaggerClient'
end
```
3) Run `pod install`
4) Build project

##### Suggest a Fix

Don't generate same properties in subclasses

**Comment by Dreddik** [comment]
Fixed in 2.2.2

**Comment by alexandern** [comment]
See issue
[3829](https://github.com/swagger-api/swagger-codegen/issues/3829)
